src/components/DropZone/components/FileUpload/FileUpload.tsx

```diff
@@ -80,71 +80,73 @@ export class FileUpload extends React.Component<CombinedProps, State> {
   }
 
   render() {
+    const fileUploadMarkup = this.renderFileUploadMarkup();
+
+    return <div className={styles.FileUpload}>{fileUploadMarkup}</div>;
+  }
+
+  private renderFileUploadMarkup() {
     const {
-      context: {size, type},
+      context: {width, height, type},
     } = this.props;
+
     const {actionTitle, actionHint} = this.state;
     const imageClasses = classNames(
       styles.Image,
-      size && size === 'extraLarge' && styles.sizeExtraLarge,
-      size && size === 'large' && styles.sizeLarge,
+      width && width === Size.ExtraLarge && styles.sizeExtraLarge,
+      width && width === Size.Large && styles.sizeLarge,
     );
 
-    const extraLargeView =
-      size === 'extraLarge' ? (
-        <Stack vertical>
-          {type === 'file' && (
-            <img className={imageClasses} src={AssetFileUpload} alt="" />
-          )}
-          {type === 'image' && (
-            <img className={imageClasses} src={AssetImageUpload} alt="" />
-          )}
-          <Button>{actionTitle}</Button>
-          <TextStyle variation="subdued">{actionHint}</TextStyle>
-        </Stack>
-      ) : null;
-
-    const largeView =
-      size === 'large' ? (
-        <Stack vertical spacing="tight">
-          {type === 'file' && (
-            <img className={imageClasses} src={AssetFileUpload} alt="" />
-          )}
-          {type === 'image' && (
-            <img className={imageClasses} src={AssetImageUpload} alt="" />
-          )}
-          <Button size="slim">{actionTitle}</Button>
-          <Caption>
+    const size = Math.min(width, height);
+    switch (size) {
```

Comment:
Just curious why you used a switch case here?

Reply:
I think it makes sense, the 4 layouts are different enough. I'd probably leave the Size.Small as default just in case.

Reply:
We don't need to encase the switch statement in a block 😄. To Dan we should be ok the way width and height are typed, if we wanted to be safe we could always create a min function that returns a type safe variable. Or return null by default? 🤷‍♂️